src/test/java/uk/gov/justice/log/integration/SearchServiceIT.java

```diff
@@ -0,0 +1,681 @@
+import static com.jayway.jsonpath.matchers.JsonPathMatchers.withJsonPath;
+import static org.apache.http.entity.ContentType.APPLICATION_JSON;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static uk.gov.justice.log.utils.CommonConstant.ELASTIC_SEARCH_CLUSTER_URL;
+import static uk.gov.justice.log.utils.CommonConstant.MAX_RETRY_TIMEOUT_MILLIS;
+import static uk.gov.justice.log.utils.CommonConstant.POST;
+import static uk.gov.justice.log.utils.CommonConstant.PUT;
+
+import uk.gov.justice.log.search.KibanaQueryBuilder;
+import uk.gov.justice.log.search.SearchCriteria;
+import uk.gov.justice.log.search.SearchService;
+import uk.gov.justice.log.utils.ConnectionManager;
+import uk.gov.justice.log.utils.PropertyReader;
+import uk.gov.justice.log.utils.RestConfig;
+import uk.gov.justice.log.utils.SearchParameters;
+import uk.gov.justice.log.wrapper.HttpAsyncClientBuilderWrapper;
+import uk.gov.justice.log.wrapper.RequestConfigBuilderWrapper;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpHost;
+import org.apache.http.nio.entity.NStringEntity;
+import org.apache.http.util.EntityUtils;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.common.io.FileSystemUtils;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.node.Node;
+import org.elasticsearch.node.NodeValidationException;
+import org.elasticsearch.node.internal.InternalSettingsPreparer;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.transport.Netty4Plugin;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.runners.MethodSorters;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.yaml.snakeyaml.Yaml;
+
+
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class SearchServiceIT {
+
+    private static final String HOST_SCHEME = "http";
+    private static final String HOST_NAME = "localhost";
+    private static final int HOST_PORT = 9205;
+    private static final String SAMPLE_LOGS_PATH = "src/test/resources/products.json";
+    private static final String HTTP_TRANSPORT_PORT = "9307";
+    private static final String ES_WORKING_DIR = "target/es";
```

Comment:
Why not create a temp directory using Files.createTempDirectory rather than polluting the target?